modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java

```diff
@@ -133,29 +157,50 @@ public void execute()
             int failCount = 0;
 
             ProjectResultReport response;
-            for (String file : files) {
-                String fileName = file.toLowerCase();
-
-                File f = new File(recipeDirectory, file);
-
-                if (fileName.endsWith(".json")) {
-                    recipeCount++;
-                    response = runJsonRecipe(f);
-                } else if (fileName.endsWith(".xml")) {
-                    projectCount++;
-                    response = runXmlProject(f);
-                } else {
-                    getLog().warn("Unexpected filename: " + fileName);
-                    continue;
-                }
+            //Run recipes
```

Comment:
I tried to extract method, but that will require much more refactoring. So, I gave up, probably we should do it in a separate PR.